src/components/Accounts/Actions/WaiveModal.js

```diff
@@ -84,38 +85,96 @@ class WaiveModal extends React.Component {
       });
       this.props.initialize({ waive: parseFloat(selected).toFixed(2), notify: true });
     }
-    return (this.props.accounts !== nextProps.accounts
-     || this.state !== nextState ||
-     this.props.open !== nextProps.open ||
-     this.props.pristine !== nextProps.pristine ||
-     this.props.invalid !== nextProps.invalid);
+    return (
+      this.props.accounts !== nextProps.accounts
+      || this.state !== nextState
+      || this.props.open !== nextProps.open
+      || this.props.pristine !== nextProps.pristine
+      || this.props.invalid !== nextProps.invalid
+    );
   }
 
   onChangeWaive(e) {
-    this.setState({ waive: e.target.value });
+    if (e.target.value === '') {
+      this.setState({ waive: 0.00 });
+    } else {
+      this.setState({ waive: parseFloat(e.target.value) });
+    }
   }
 
+  calculateSelectedAmount() {
+    const { accounts } = this.props;
+    const result = accounts.reduce((selected, { remaining }) => {
+      return selected + parseFloat(remaining);
+    }, 0);
+    return result;
+  }
+
+  renderFormMessage() {
+    const {
+      accounts = []
+    } = this.props;
+
+    const selected = this.calculateSelectedAmount();
+    const waiveAmount = this.state.waive === ''
+      ? 0.00
+      : this.state.waive;
+
+    const waiveType = this.state.waive < selected
+      ? <FormattedMessage id="ui-users.accounts.waive.summary.partiallyWaive" />
+      : <FormattedMessage id="ui-users.accounts.waive.summary.waiving" />;
+
+    const feeFineForm = accounts.length === 1
+      ? <FormattedMessage id="ui-users.accounts.feeFine" />
+      : <FormattedMessage id="ui-users.accounts.feesFines" />;
+
+    return (
+      <FormattedMessage
+        id="ui-users.accounts.waive.summary"
+        values={{
+          feesFinesAmount: accounts.length,
+          waiveAmount: parseFloat(waiveAmount).toFixed(2),
+          waiveType,
+          feeFineForm
+        }}
+      />
+    );
+  }
+
+  onCancel = () => {
+    this.props.onClose();
+    this.props.reset();
+  };
+
+  onSubmit = () => {
+    this.props.handleSubmit();
+    this.props.reset();
+  };
```

Comment:
could u destructure props in all methods `onCancel` and `onSubmit` that u've changed